server/jetty/src/main/java/io/deephaven/server/jetty/jsplugin/JsPlugins.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.server.jetty.jsplugin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.deephaven.plugin.type.JsPlugin;
+import io.deephaven.plugin.type.JsPluginRegistration;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link JsPluginRegistration} for Jetty.
+ *
+ * @see #servletHolder(String)
+ */
+public final class JsPlugins implements JsPluginRegistration {
+
+    private static final String PLUGINS = "plugins";
+    private static final String NAME = "name";
+    private static final String VERSION = "version";
+    private static final String MAIN = "main";
+    private static final String MANIFEST_JSON = "manifest.json";
+
+    /**
+     * Creates a new js plugins instance with a temporary zip filesystem.
+     *
+     * @return the js plugins
+     * @throws IOException if an I/O exception occurs
+     */
+    public static JsPlugins create() throws IOException {
+        final Path tempDir = Files.createTempDirectory(JsPlugins.class.getName());
+        tempDir.toFile().deleteOnExit();
+        final Path fsZip = tempDir.resolve("fs.zip");
+        fsZip.toFile().deleteOnExit();
+        // Note, the URI needs explicitly be parseable as a directory URL ending in "!/", a requirement of the jetty
+        // resource creation implementation, see
+        // org.eclipse.jetty.util.resource.Resource.newResource(java.lang.String, boolean)
+        final URI uri = URI.create(String.format("jar:file:%s!/", fsZip));
+        final JsPlugins jsPlugins = new JsPlugins(uri);
+        jsPlugins.init();
+        return jsPlugins;
+    }
+
+    private final URI filesystem;
+    private final List<Map<String, String>> plugins;
+
+    private JsPlugins(URI filesystem) {
```

Comment:
you were going to add tests for this, others?

Reply:
Given the test construction right now, it's _easy_ for me to test the empty case; and at a minimum, that's testing that we correctly wrote out and read back the manifest.json file from the zip filesystem.

It probably makes sense to eventually add more HTTP tests against io.deephaven.server.jetty.JettyBackedGrpcServer that will check our expectations (etag, caching, etc), and have easier injectability for JsPlugin and other HTTP stuff we might want to test.